ipc/src/server.rs

```diff
@@ -36,13 +36,25 @@ impl<M: Metadata, S: Middleware<M>> tokio_service::Service for Service<M, S> {
 	type Request = String;
 	type Response = Option<String>;
 
-	type Error = ();
+	type Error = std::io::Error;
 
-	type Future = FutureResult<S::Future>;
+	type Future = Box<Future<Item=Self::Response,Error=Self::Error> + Send>;
 
 	fn call(&self, req: Self::Request) -> Self::Future {
 		trace!(target: "ipc", "Received request: {}", req);
-		self.handler.handle_request(&req, self.meta.clone())
+		let fut = self.handler
+			.handle_request(&req, self.meta.clone())
+			.then(|result| {
+				match result {
+					Err(_) => {
```

Comment:
Maybe log a warning here. Why not propagate the error upstream?

Reply:
It's because the error type here is `()`. Should the server shut down if a handler invocation on the `MetaIoHandler` fails? If so, we can propagate here. What kind of error should it be here then?

Reply:
I think it's non-failing anyway, so it's good.